src/Rector/Class_/CommandDescriptionToPropertyRector.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Symfony\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Property;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://symfony.com/blog/new-in-symfony-5-3-lazy-command-description
+ *
+ * @see \Rector\Symfony\Tests\Rector\Class_\CommandDescriptionToPropertyRector\CommandDescriptionToPropertyRectorTest
+ */
+final class CommandDescriptionToPropertyRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Symfony Command description setters are moved to properties', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Symfony\Component\Console\Command\Command
+
+final class SunshineCommand extends Command
+{
+    protected static $defaultName = 'sunshine';
+    public function configure()
+    {
+        $this->setDescription('sunshine description');
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use Symfony\Component\Console\Command\Command
+
+final class SunshineCommand extends Command
+{
+    protected static $defaultName = 'sunshine';
+    protected static $defaultDescription = 'sunshine description';
+    public function configure()
+    {
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isObjectType($node, new ObjectType('Symfony\Component\Console\Command\Command'))) {
+            return null;
+        }
+
+        $defaultNameProperty = $node->getProperty('defaultDescription');
+        if ($defaultNameProperty instanceof Property) {
+            return null;
+        }
+
+        $commandDescription = $this->resolveCommandDescription($node);
+        if (! $commandDescription instanceof Node) {
+            return null;
+        }
+
+        if (! $commandDescription instanceof String_ && ! $commandDescription instanceof ClassConstFetch) {
```

Comment:
ClassConstFetch can be dynamic value, you can verify with ExprAnalyzer, see https://github.com/rectorphp/rector-src/blob/73aa7e3d68d74fc0b9b26fc543bf1207cfc8d0dd/src/NodeAnalyzer/ExprAnalyzer.php#L69

Reply:
I dont realy understand what i have to do. Are you saying i have to check that the `$commandDescription` is `isDynamicExpr` otherwise return null? like:
```suggestion
        if (! $commandDescription instanceof String_ && ! $commandDescription instanceof ClassConstFetch && ! $this->exprAnalyzer->isDynamicExpr($commandDescription)) {
```


I used this code as example btw https://github.com/rectorphp/rector-symfony/blob/main/src/Rector/Class_/MakeCommandLazyRector.php#L97 where the same check is done.

Reply:
You should can use return null early when it is dynamic:

```php

if (! $commandDescription instanceof Expr) {
    return null;
}

if ($this->exprAnalyzer->isDynamicExpr($commandDescription)) {
    return null;
}
```

You need to inject `Rector\Core\NodeAnalyzer\ExprAnalyzer` to constructor first for that:

```php
public function __construct(private readonly \Rector\Core\NodeAnalyzer\ExprAnalyzer $exprAnalyzer)
{
}
```

Reply:
Check i added this check and also removed check on Node as that was doing nothing anymore

Reply:
String_ and ClassConstFetch check no longer needed, can be removed.